### DISCUSSION TITLE
indicator in simple_clock, when timer(_face) is running

### DISCUSSION THREAD

**Discussion by 814d3** [joeycastillo/Sensor-Watch#366] [Ideas]
Hi all, especially @TheOnePerson,

I use the timer_face quite often in background (simple_clock ist active while the timer counts down several hours/mins) and would like to have an indicator, that a timer is running. For example the LAP indicator could be active or flash while a timer is active. Maybe flashing would be a good idea as the low voltage warning will activate a constant LAP signal - so it should be easily distinguishable.

Greetings

**Comment by WesleyBlack** [comment]
Hi 814d3,

This is one of the first things I did to my watch. You have to modify four files (movement.h, movement.c, timer_face.c, and simple_clock_face.c) but it's a fairly straight forward process. It seems the reason this wasn't done is because the bell icon that the timer face uses when active is used to indicate if the hourly chime turned on. I don't use the hourly chime so I repurposed the bell icon on the simple clock face to indicate that a timer is running in the background. Below is the code I used to achieve this:

**movement.h**
Added new line above 49: `bool timer_running : 1;       // if true, displays bell on simple_clock_face.`

**movement.c**
Added new line above 383: `movement_state.settings.bit.timer_running = false;`

**timer_face.c**
Added new line above 269: `settings->bit.timer_running = false;`
Added new line above 275: `settings->bit.timer_running = true;`
Added new line below 282: `settings->bit.timer_running = true;`
Added new line below 308: `settings->bit.timer_running = false;`

**simple_clock_face.c**
Changed `if (state->signal_enabled)` on line 57 to `if (settings->bit.timer_running)`

Hope this helps 🙂

**Reply by 814d3** [reply]
I implemented your code changes, unfortunately they're not running as expected. One thing that I've noticed so far is that the bell icon is also activated in simple_clock_face when I just scroll through the presets, while it's not activated in the timer_face itself. Moreover, when I start a timer (bell icon active in timer_face) and stop it afterwards (bell icon deactivated in timer_face), it is still active in simple_clock_face. 
I've no time right now to investigate further, but wanted to tell you. Anyway, thank you so far.

**Reply by WesleyBlack** [reply]
Did you end up figuring it out? I changed what the buttons do on my version, so I forgot exactly how the timer functioned before my modifications.

Try adding the following code to **timer_face.c**
Add above 330: `settings->bit.timer_running = true;`
Add above 334: `settings->bit.timer_running = false;`

**Reply by 814d3** [reply]
I haven't had the time yet. But I think I would rather use the "PM" indicator instead of the bell or the lap-sign. Sometimes i use the hourly signal and the lap sign is also used within timer_face...

**Comment by 814d3** [comment]
Hi WesleyBlack,

thanks a lot for the code. I'll get into it and try to activate the LAP indicator instead. But before, I hope somebody can fix the bug I experience with the timer_face [#377](https://github.com/joeycastillo/Sensor-Watch/issues/377). For me it is very unreliable and I don't use the timer right now.

**Reply by WesleyBlack** [reply]
Interesting I never had an issue with the the timer. 

You can have it activate the LAP indicator instead of the Bell by adding the follow code above line 56 on **simple_clock_face.c** instead of the changed code I provided above
`if (settings->bit.timer_running) watch_set_indicator(WATCH_INDICATOR_LAP);
    else watch_clear_indicator(WATCH_INDICATOR_LAP);`

You'll also want to remove line 96 which turns on the LAP indicator when the battery is low.

**Reply by 814d3** [reply]
Thanks!
Strange, maybe I need to flash my watch again...
Btw. #372 hasn't occurred for you either?

**Reply by WesleyBlack** [reply]
Sorry I can't really speak to that issue. I only keep two presets timers saved on my timer face
